plcc/src/proteinstructure/Molecule.java

```diff
@@ -157,12 +151,90 @@ public String getAltLocWithMostAtoms() {
+                DP.getInstance().w("Chosen altLoc '" + chosenAltLoc + "' leads to " + numAtomsWithChosenAltLoc + " atoms for AA molecule " + this.getFancyName() + ".");
+            }
+        }
+        deletedAtoms = this.deleteAtomsWithAltLocDifferentFrom(chosenAltLoc);
+                
+        if(this.atoms.size() < 1) {
+            DP.getInstance().w("Molecule  " + this.getFancyName() + " of chain " + this.getChainID() + " has no atoms after choosing alternative location PDB field (had " + numAtomsBefore + " before).");
+        }
+        
+        return deletedAtoms;
+    }
+    
     
     /**
      * Determines the center atom of this molecule, and also sets the center sphere radius for the molecule.
+     * There is an override function for residues in the Residue class that returns the Calpha as the center atom.
      * @return the center atom
      */
-    abstract public Atom getCenterAtom();
+    public Atom getCenterAtom(){
+       
+        Atom a, b, center = null;
+        Integer maxDistForAtom, dist = 0;       // just assign a small start value
+        Integer MAXDIST = Integer.MAX_VALUE;    // just assign a *very* large start value
+        Integer totalMinMaxDist = MAXDIST;
+        
+        if(atoms.size() < 1) {
+            if( ! Settings.getBoolean("plcc_B_no_parse_warn")) {
+                DP.getInstance().w("getCenterAtom(): PDB molecule  " + this.pdbNum + " chain " + this.getChainID() + " of type " + getName3() + " has " + atoms.size() + " atoms in default location, returning null.");
+            }
+            return(null);
+        }
+        // The center is calculated by finding the atom with the minimal maximal distance to all other atoms of this molecule.
+        for(Integer i = 0; i < atoms.size(); i++) {
+            a = atoms.get(i);
+            maxDistForAtom = 0;
+
+            for(Integer j = 0; j < atoms.size(); j++){
+                b = atoms.get(j);
+                dist = a.distToAtom(b);
+
+                if(dist > maxDistForAtom){
+                    maxDistForAtom = dist;
+                }
+            }
+            
+            // If the maxDist of the checked atom is smaller than the current maxDist, the center atom is updated.
+            if(maxDistForAtom < totalMinMaxDist) {
+                totalMinMaxDist = maxDistForAtom;
+                center = a;
+            }
+        }
+
+        this.centerSphereRadius = totalMinMaxDist;
+        
+        // If totalMinMaxDist still has the original value of MAXDIST something must be wrong.
+        if(Objects.equals(totalMinMaxDist, MAXDIST)) {
+            DP.getInstance().w("OR: MinMax distance of the atoms of PDB molecule "+ pdbNum + " is >= " + MAXDIST + ", seems *very* unlikely.");
+        }
+
+        if(center == null) {
+            DP.getInstance().w("ERROR: Could not determine center atom of molecule  type " + this.getType() + " with PDB number " + pdbNum + ", DSSP number " + dsspNum + ". Returning null.");
```

Comment:
Throwing a warning beginning with 'ERROR' seems confusing. Note that DP's w() and e() start their print with '[WARNING]' and '[ERROR]', respectively.